src/main/java/org/openstreetmap/atlas/generator/AtlasGeneratorHelper.java

```diff
@@ -172,6 +186,92 @@ public String getName()
         };
     }
 
+    public static Set<LineString> convertMultiPolygonToLineCollection(
+            final MultiPolygon multipolygon)
+    {
+        final Set<LineString> linestrings = new HashSet<>();
+        for (int i = 0; i < multipolygon.getNumGeometries(); i++)
+        {
+            final Polygon part = (Polygon) multipolygon.getGeometryN(i);
+            linestrings.add(part.getExteriorRing());
+            for (int j = 0; j < part.getNumInteriorRing(); j++)
+            {
+                linestrings.add(part.getInteriorRingN(j));
+            }
+        }
+        return linestrings;
+    }
+
+    /**
+     * This function takes a given tuple of shard and atlas to the set of multipolygons that were
+     * assembled for it. It will then apply each of those relation multipolygons as feature changes
+     * and save the final atlas with the geometries serialized
+     *
+     * @return function that does the above
+     */
+    protected static PairFunction<Tuple2<String, Tuple2<Atlas, org.apache.spark.api.java.Optional<Set<Tuple2<Long, MultiPolygon>>>>>, String, Atlas> augmentAtlas()
+    {
+        return (Serializable & PairFunction<Tuple2<String, Tuple2<Atlas, org.apache.spark.api.java.Optional<Set<Tuple2<Long, MultiPolygon>>>>>, String, Atlas>) tuple ->
+        {
+            final Atlas rawAtlas = tuple._2._1;
+            if (!tuple._2()._2.isPresent())
```

Comment:
With just the empty set, here you could just check that the set is empty or not...